File: InterviewStreet/stringsimilarity/StringSimilarity.py

def SimilarStrings(string):
    reverseString  = string[::-1]
    tempStr = ''
    total = 0
    for s in reverseString:
        tempStr = s + tempStr
        count = 0
        for i in tempStr:
            if string[count] == tempStr[count]:
                count = count + 1
            else:
                break
        total = total + count 
    return total

N = int(raw_input())
for i in xrange(N):
    string = str(raw_input())
    print SimilarStrings(string)

File: InterviewStreet/stringsimilarity/stringsimilarity.cpp
#include<iostream>

using namespace std;

int similarstrings(char *string, int index){
    int  i;
    for(i=index; string[i]!= '\0' ; i++){
        if(string[i] != string [i-index]){
           return i-index; 
        }
    }
    return i-index;
}

int main(){
    int no;
    cin >> no;
    char string[100001];
    for(int i=0; i<no; i++){
        cin >> string;
        int total = 0;
        for(int j=0; string[j]!= '\0'; j++){
            total = total + similarstrings(string, j);
        } 
        cout << total << endl;
    }
}
